eZ/Publish/Core/settings/repository.yml

```diff
@@ -63,88 +76,220 @@ services:
+    ezpublish.api.service.inner_section:
+        class: %ezpublish.api.service.section.class%
         factory: ["@ezpublish.api.repository.factory", buildService]
-        arguments: ["@ezpublish.api.repository", Section]
+        arguments: ["@ezpublish.api.inner_repository", Section]
         lazy: true
 
-    ezpublish.api.service.trash:
-        class: "%ezpublish.api.service.trash.class%"
+    ezpublish.api.service.inner_trash:
+        class: %ezpublish.api.service.trash.class%
         factory: ["@ezpublish.api.repository.factory", buildService]
-        arguments: ["@ezpublish.api.repository", Trash]
+        arguments: ["@ezpublish.api.inner_repository", Trash]
         lazy: true
 
-    ezpublish.api.service.location:
-        class: "%ezpublish.api.service.location.class%"
+    ezpublish.api.service.inner_location:
+        class: %ezpublish.api.service.location.class%
         factory: ["@ezpublish.api.repository.factory", buildService]
-        arguments: ["@ezpublish.api.repository", Location]
+        arguments: ["@ezpublish.api.inner_repository", Location]
         lazy: true
 
-    ezpublish.api.service.language:
-        class: "%ezpublish.api.service.language.class%"
+    ezpublish.api.service.inner_language:
+        class: %ezpublish.api.service.language.class%
         factory: ["@ezpublish.api.repository.factory", buildService]
-        arguments: ["@ezpublish.api.repository", ContentLanguage]
+        arguments: ["@ezpublish.api.inner_repository", ContentLanguage]
         lazy: true
 
     # Signal Slot API wrapper
     ezpublish.signalslot.repository:
-        class: "%ezpublish.signalslot.repository.class%"
-        arguments: ["@ezpublish.api.inner_repository", "@ezpublish.signalslot.signal_dispatcher_transaction_wrapper"]
-        lazy: true
+        class: '%ezpublish.signalslot.repository.class%'
+        arguments:
+            - '@ezpublish.api.inner_repository'
+            - '@ezpublish.signalslot.signal_dispatcher_transaction_wrapper'
+            - '@ezpublish.signalslot.service.content'
+            - '@ezpublish.signalslot.service.content_type'
+            - '@ezpublish.signalslot.service.field_type'
+            - '@ezpublish.signalslot.service.role'
+            - '@ezpublish.signalslot.service.object_state'
+            - '@ezpublish.signalslot.service.url_wildcard'
+            - '@ezpublish.signalslot.service.url_alias'
+            - '@ezpublish.signalslot.service.user'
+            - '@ezpublish.signalslot.service.search'
+            - '@ezpublish.signalslot.service.section'
+            - '@ezpublish.signalslot.service.trash'
+            - '@ezpublish.signalslot.service.location'
+            - '@ezpublish.signalslot.service.language'
+
+    ezpublish.signalslot.service.content:
+        class: '%ezpublish.signalslot.service.content.class%'
+        arguments:
+            - '@ezpublish.api.service.inner_content'
+            - '@ezpublish.signalslot.signal_dispatcher'
```

Comment:
This should be `ezpublish.signalslot.signal_dispatcher_transaction_wrapper` (on SS services below as well)